### PR TITLE
Mon 3616 async

### PR DESCRIPTION
**Pull request by bouda1**
Le stream connector is asynchronous now. If a script is too long to execute, messages are sent by broker but it should not slow down it.

### CODE REVIEW

**Comment by bouda1** [comment]
There is a part about filters that is commented. I hope to come back on it very soon... :smile:

File: lua/inc/com/centreon/broker/lua/stream.hh [hunk 1]
```diff
@@ -46,13 +87,10 @@ class stream : public io::stream {
...
   bool read(std::shared_ptr<io::data>& d, time_t deadline) override;
   int write(std::shared_ptr<io::data> const& d) override;
-
- private:
-  // Access to the Lua interpreter
-  luabinding* _luabinding;
-
-  // Cache
-  macro_cache _cache;
+//  bool filter(uint32_t type);
```
Comment (SylvestreG) [inline lua/inc/com/centreon/broker/lua/stream.hh]: you can remove this comment.

File: lua/src/stream.cc [hunk 2]
```diff
@@ -77,8 +179,75 @@ int stream::write(std::shared_ptr<io::data> const& data) {
...
+//  std::unique_lock<std::mutex> loop_lock(_loop_m);
+//  _filter = type;
+//  _loop_cv.notify_one();
+//  loop_lock.unlock();
+//
+//  // Wait for the answer
+//  std::unique_lock<std::mutex> ans_lock(_filter_m);
+//  _filter_cv.wait(ans_lock, [this] { return _filter == 0; });
+//  return _filter_ok;
+//}
```
Comment (SylvestreG) [inline lua/src/stream.cc]: you can delete this code too

File: lua/src/stream.cc [hunk 3]
```diff
@@ -41,15 +45,113 @@ using namespace com::centreon::broker::lua;
...
+        {
+          std::lock_guard<std::mutex> lock(_acks_count_m);
+          _acks_count += res;
+        }
+        lock.lock();
+      }
+    }
+
+//    // Unregister the filter
+//    multiplexing::muxer::unregister_read_filter();
```
Comment (SylvestreG) [inline lua/src/stream.cc]: same here

File: lua/src/stream.cc [hunk 4]
```diff
@@ -41,15 +45,113 @@ using namespace com::centreon::broker::lua;
...
+//      if (_filter) {
+//        lock.unlock();
+//        {
+//          std::lock_guard<std::mutex> ans_lock(_filter_m);
+//          _filter_ok = lb->filter(_filter);
+//          _filter = 0;
+//          _filter_cv.notify_one();
+//        }
+//        lock.lock();
+//      }
```
Comment (SylvestreG) [inline lua/src/stream.cc]: same here

File: lua/src/stream.cc [hunk 5]
```diff
@@ -41,15 +45,113 @@ using namespace com::centreon::broker::lua;
...
+  _thread = std::thread([&] {
+    // Access to the Lua interpreter
+    luabinding* lb = nullptr;
+
+    // If there is a filter, register it.
+//    if (lb->has_filter()) {
+//      std::function<bool(uint32_t)> func =
+//          std::bind(&stream::filter, this, std::placeholders::_1);
+//      multiplexing::muxer::register_read_filter(func);
+//    }
```
Comment (SylvestreG) [inline lua/src/stream.cc]: same here

File: core/inc/com/centreon/broker/misc/math.hh [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+CCB_BEGIN()
+
+namespace misc {
+/**
+ *  Compute a and b such that y=ax+b is a linear regression of the pts array
+ *  given in parameter. The method used is least squares.
+ *
+ * @param pts An array of pairs (x, y)
+ * @param size If shorter than the array size, only the first size pairs of
+ *             the array are taken for the regression regression.
```
Comment (agelibert) [inline core/inc/com/centreon/broker/misc/math.hh]: typo regression x2

File: core/inc/com/centreon/broker/misc/math.hh [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  double cov_xy = sxy / count - mx * my;
+  double sd_x2 = sx2 / count - mx * mx;
+
+  if (sd_x2)
+    a = cov_xy / sd_x2;
+  else
+    return false;
+  b = my - a * mx;
+
+  return true;
```
Comment (agelibert) [inline core/inc/com/centreon/broker/misc/math.hh]: Better shape ?

```
// Postcondition
if (!sd_x2) 
    return false;

a = cov_xy / sd_x2;
b = my - a * mx;
return true;
```

**Pull request merged**